tfexec/terraform_test.go

```diff
@@ -125,7 +126,155 @@ resource "aws_security_group" "bar" {}
 	}
 
-	switchBackToRemotekFunc, err := terraformCLI.OverrideBackendToLocal(context.Background(), filename, workspace, false)
+	switchBackToRemotekFunc, err := terraformCLI.OverrideBackendToLocal(context.Background(), filename, workspace, false, nil)
+	if err != nil {
+		t.Fatalf("failed to run OverrideBackendToLocal: %s", err)
+	}
+
+	if _, err := os.Stat(filepath.Join(terraformCLI.Dir(), filename)); os.IsNotExist(err) {
+		t.Fatalf("the override file does not exist: %s", err)
+	}
+
+	updatedState, _, err := terraformCLI.StateMv(context.Background(), state, nil, "aws_security_group.foo", "aws_security_group.foo2")
+	if err != nil {
+		t.Fatalf("failed to run terraform state mv: %s", err)
+	}
+
+	changed, err = terraformCLI.PlanHasChange(context.Background(), updatedState)
+	if err != nil {
+		t.Fatalf("failed to run PlanHasChange: %s", err)
+	}
+	if changed {
+		t.Fatalf("expect not to have changes")
+	}
+
+	switchBackToRemotekFunc()
+
+	if _, err := os.Stat(filepath.Join(terraformCLI.Dir(), filename)); err == nil {
+		t.Fatalf("the override file wasn't removed: %s", err)
+	}
+
+	changed, err = terraformCLI.PlanHasChange(context.Background(), nil)
+	if err != nil {
+		t.Fatalf("failed to run PlanHasChange: %s", err)
+	}
+	if !changed {
+		t.Fatalf("expect to have changes")
+	}
+}
+
+func TestAccTerraformCLIOverrideBackendToLocalWithBackendConfig(t *testing.T) {
+	SkipUnlessAcceptanceTestEnabled(t)
+
+	endpoint := "http://localhost:4566"
+	localstackEndpoint := os.Getenv("LOCALSTACK_ENDPOINT")
+	if len(localstackEndpoint) > 0 {
+		endpoint = localstackEndpoint
+	}
+
+	backend := fmt.Sprintf(`
+terraform {
+  # https://www.terraform.io/docs/backends/types/s3.html
+  backend "s3" {
+    region = "ap-northeast-1"
+    // bucket = "tfstate-test"
+    key    = "%s/terraform.tfstate"
+
+    // mock s3 endpoint with localstack
+    endpoint                    = "%s"
+    access_key                  = "dummy"
+    secret_key                  = "dummy"
+    skip_credentials_validation = true
+    skip_metadata_api_check     = true
+    force_path_style            = true
+  }
+}
+# https://www.terraform.io/docs/providers/aws/index.html
+# https://www.terraform.io/docs/providers/aws/guides/custom-service-endpoints.html#localstack
+provider "aws" {
+  region = "ap-northeast-1"
+
+  access_key                  = "dummy"
+  secret_key                  = "dummy"
+  skip_credentials_validation = true
+  skip_metadata_api_check     = true
+  skip_region_validation      = true
+  skip_requesting_account_id  = true
+  s3_force_path_style         = true
+
+  // mock endpoints with localstack
+  endpoints {
+    s3  = "%s"
+    ec2 = "%s"
+    iam = "%s"
+  }
+}
+`, t.Name(), endpoint, endpoint, endpoint, endpoint)
+	source := `
+resource "aws_security_group" "foo" {}
+resource "aws_security_group" "bar" {}
+`
+	workspace := "work1"
+	backendConfig := []string{"bucket=tfstate-test"}
+	e := SetupTestAcc(t, source)
```

Comment:
The test is failing: https://github.com/minamijoyo/tfmigrate/runs/7225213659?check_suite_focus=true

```
2022/07/07 00:45:47 [DEBUG] [executor@/tmp/workDir3305983562] failed to run command: (*exec.ExitError)(0xc0002cc640)(exit status 1)
--- FAIL: TestAccTerraformCLIOverrideBackendToLocalWithBackendConfig (3.65s)
    terraform_test.go:242: failed to run terraform apply: failed to run command (exited 1): terraform apply -input=false -no-color -auto-approve
        stdout:
        stderr:
        Error: error configuring Terraform AWS Provider: no valid credential sources for Terraform AWS Provider found.
        Please see https://registry.terraform.io/providers/hashicorp/aws
        for more information about providing credentials.
        Error: failed to refresh cached credentials, no EC2 IMDS role found, operation error ec2imds: GetMetadata, failed to get API token, cannot get API token, operation error ec2imds: getToken, http response error StatusCode: 400, request to EC2 IMDS failed
```

I think the mocked test config should be `backend+source` instead of `source`.

Reply:
Thanks, passed `source+backend` in bcae4e181cc56debdea4e69f220d74fe7e2c28fa